src/course/en/course.json

```diff
@@ -62,7 +62,9 @@
         "incorrect": "Incorrect",
         "correct": "Correct",
         "locked": "Locked",
-        "visited": "Visited"
+        "visited": "Visited",
+        "mandatory": "Mandatory",
```

Comment:
Definitely mandatory and not required?

Reply:
Mandatory was suggested by Stu and I suppose it sounds better.

Reply:
@oliverfoster , mandatory is the most commonly used but I've seen others

Reply:
Required:
https://github.com/adaptlearning/adapt-contrib-core/blob/de6fcd9c58ad7ed0d793920b048fcb213bdcc980/schema/contentobject.schema.json#L99-L107

https://github.com/adaptlearning/adapt-contrib-core/blob/de6fcd9c58ad7ed0d793920b048fcb213bdcc980/schema/article.schema.json#L11-L19

https://github.com/adaptlearning/adapt-contrib-core/blob/de6fcd9c58ad7ed0d793920b048fcb213bdcc980/schema/block.schema.json#L24-L32

https://github.com/adaptlearning/adapt-contrib-core/blob/de6fcd9c58ad7ed0d793920b048fcb213bdcc980/js/models/itemsQuestionModel.js#L88

Mandatory:
https://github.com/adaptlearning/adapt-contrib-core/blob/de6fcd9c58ad7ed0d793920b048fcb213bdcc980/js/models/adaptModel.js#L310

https://github.com/adaptlearning/adapt-contrib-branching/blob/cb9e67b5639751fa2614ddd8d16c78a94536b54b/README.md?plain=1#L38-L56

`required` is substantially more common in the open source code. The only place I've seen `required` used so far is for extra status labels and 'mandatory' is used in custom menus and diagnostics.

Where are you planning on using it? What's the use-case?

Reply:
@oliverfoster. Yeah thats cool. It's mainly for labelling 'on course' items so for use with diagnostic etc labelling content objects on page and items on the menu. So it's different to code-y names but happy to go with your suggestion.

Reply:
(nb. I meant commonly used for this use case)

Reply:
> So it's different to code-y names

I understand that about the setting value `"Mandatory"`, where the value can become rendered output, but the setting name `"mandatory":` is code-y, it would be the variable name that gets referenced in other bits of code via `_globals._accessibility.mandatory`. This would be the only place in open source that the variable `mandatory` is used (apart from grunt, where the [less compiler](https://github.com/adaptlearning/adapt_framework/blob/5663a1380dc49c486af4b4ab7dd368ac66c7ee18/grunt/config/less.js#L51) and [copy task](https://github.com/adaptlearning/adapt_framework/blob/5663a1380dc49c486af4b4ab7dd368ac66c7ee18/grunt/config/copy.js#L49) use it for task importance).

> happy to go with your suggestion

The suggestion for consistency and uniformity would be `"required": "Required"`.

However, I do like the idea of making course content optionality split over the enums `mandatory` and `optional` as it creates a clear distinction between the multiple uses of `required` in libraries, schemas and sundry other configs. It is nice to differentiate concepts.

Reply:
@oliverfoster agreed